### PR TITLE
[15.0][MIG]hr_applicant:Migrated V14 to V15

### PR DESCRIPTION
**Pull request by RohanChristianSCS**

### CODE REVIEW

**Comment by JayVora-SerpentCS** [comment]
@pragatipatel-serpentcs please review

File: hr_applicant/LICENSE/LICENSE [hunk 1]
```diff
@@ -0,0 +1,16 @@
+    OpenERP, Open Source Management Solution
+    Copyright (C) 2022-Today Serpent Consulting Services Pvt. Ltd.
```
Comment (pragatipatel-serpentcs) [inline hr_applicant/LICENSE/LICENSE]: Please update 2023

File: hr_applicant/README.rst [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+=====
+
+Bug Tracker
+===========
+
+Credits
+=======
+
+Contributors
+------------
```
Comment (pragatipatel-serpentcs) [inline hr_applicant/README.rst]: Good to have same decors '====='

File: hr_applicant/__init__.py [hunk 3]
```diff
@@ -0,0 +1,4 @@
+# See LICENSE file for full copyright and licensing details.
+
+from . import models
+from . import wizard
```
Comment (pragatipatel-serpentcs) [inline hr_applicant/__init__.py:4]: add new line at end of file

File: hr_applicant/models/__init__.py [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# See LICENSE file for full copyright and licensing details.
+from . import hr_recruitment
+from . import hr_app_med_occ
+from . import hr_app_rel_edu
+from . import hr_app_trav_lang
+from . import hr_recruitment_employee
+from . import hr_emp_med_occ
+from . import hr_emp_rel_edu
+from . import hr_emp_trav_lang
+from . import training
```
Comment (pragatipatel-serpentcs) [inline hr_applicant/models/__init__.py:10]: add line at end of file 
please check in other files also.

**Comment by JayVora-SerpentCS** [comment]
@RohanChristianSCS Pls see

**Comment by JayVora-SerpentCS** [comment]
@malaykhamar-serpentcs

File: hr_applicant/README.rst [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+* This module helps you to keep employee information.
+
+* Have subsections in Applicant and Employee forms to have several informations like Medical details, Relatives, Previous Occupations etc.
+
+* Informations which will filled for Applicant will be added automatically in Employee when applicant will converted into Employee.
+
+* This application allows you to keep description for details of Applicant and Employee like Attachment of Subsections,Medical
+
+  Details,Previous Occupation,Education Details,Relatives,Languages,Pervious Travel and Profile Report
```
Comment (MalayKhamar-SerpentCS) [inline hr_applicant/README.rst]: Please correct the "previous" spelling. (Pervious Travel)

File: hr_applicant/static/description/index.html [hunk 6]
```diff
@@ -0,0 +1,459 @@
...
+        <div class="oe_span12" style="margin-top:30px">
+            <img class="img-border oe_screenshot" src="Language.png" style="width:730px">
+        </div>
+        <div class="oe_span12" style="margin-top:40px;">
+            <center>
+                <h4 class="oe_slogan" style="font-size:28px;">
+                    <b>
+                        <span class="label" style="background:#875A7B;">
+                            <span class="fa fa-user"/>
+                                Pervious Travel
```
Comment (MalayKhamar-SerpentCS) [inline hr_applicant/static/description/index.html]: Please correct the "previous" spelling.

File: hr_applicant/models/training.py [hunk 7]
```diff
@@ -0,0 +1,289 @@
...
+                _("End Date should be greated than Start date of Training!")
+            )
+
+    course_id = fields.Many2one("training.courses", string="Course Name", required=True)
+    department = fields.Char(
+        related="course_id.department", string="Department", readonly=True
+    )
+    job_id = fields.Many2one(
+        related="course_id.job_id",
+        comodel="hr.job",
```
Comment (MalayKhamar-SerpentCS) [inline hr_applicant/models/training.py]: Wrong Attribute comodel. Change it to "comodel_name".

File: hr_applicant/models/training.py [hunk 8]
```diff
@@ -0,0 +1,289 @@
...
+        string="Applied Job",
+        readonly=True,
+    )
+    course_categ_id = fields.Many2one(
+        related="course_id.course_type_id",
+        comodel="course.type",
```
Comment (MalayKhamar-SerpentCS) [inline hr_applicant/models/training.py]: Wrong Attribute comodel. Change it to "comodel_name".

File: hr_applicant/models/hr_app_med_occ.py [hunk 9]
```diff
@@ -0,0 +1,114 @@
+from logging import warning
```
Comment (MalayKhamar-SerpentCS) [inline hr_applicant/models/hr_app_med_occ.py]: Please remove unused Importation : warning

File: hr_applicant/models/training.py [hunk 10]
```diff
@@ -0,0 +1,289 @@
...
+    _description = "Training Class"
+
+    @api.constrains("training_start_date", "training_end_date")
+    def _check_training_dup(self):
+        self.ensure_one()
+        if self.training_start_date < fields.Date.context_today(self):
+            raise ValidationError(_("You can't create past training!"))
+        if self.training_start_date > self.training_end_date:
+            raise ValidationError(
+                _("End Date should be greated than Start date of Training!")
```
Comment (MalayKhamar-SerpentCS) [inline hr_applicant/models/training.py]: Please correct the spelling of "greater".

File: hr_applicant/models/training.py [hunk 11]
```diff
@@ -0,0 +1,289 @@
...
+    )
+    description = fields.Text("Description")
+
+    @api.onchange("training_start_date", "course_id")
+    def _onchange_start_date(self):
+        for rec in self:
+            if rec.training_start_date and rec.course_id:
+                end_date = False
+                if rec.course_id.duration and rec.course_id.duration_type == "day":
+                    end_date = rec.training_start_date + datetime.timedelta(
```
Comment (MalayKhamar-SerpentCS) [inline hr_applicant/models/training.py]: Please check the wronlgy use timedelta. It should import directly from the main datetime library, not from the sub class.
Import Like : 
          from datetime import datetime, timedelta

And then use timedelta. 
Also You have already used the relativedelta , so better we use relativedelta and remove to call additional timedelta.

File: hr_applicant/models/training.py [hunk 12]
```diff
@@ -0,0 +1,289 @@
...
+            if rec.training_start_date < fields.Date.context_today(rec):
+                raise ValidationError(_("You can't create past training!"))
+            if rec.training_start_date > rec.training_end_date:
+                raise ValidationError(
+                    _("End Date should be greater than Start date of Training!")
+                )
+            if rec.date_of_arrival and rec.date_of_arrival < rec.training_start_date:
+                raise ValidationError(
+                    _(
+                        "Arrival Date should be less or equal than \
```
Comment (MalayKhamar-SerpentCS) [inline hr_applicant/models/training.py]: Do not break the warnings with \  , It adds the unwanted spaces.Try with " 
like: 
                raise ValidationError(
                    _(
                        "Arrival Date should be less or equal than "
                        "Start date of Training!"
                    )
                )

File: hr_applicant/models/training.py [hunk 13]
```diff
@@ -0,0 +1,289 @@
...
+    def _check_training_dup(self):
+        # self.ensure_one()
+        for rec in self:
+            if rec.training_start_date < fields.Date.context_today(rec):
+                raise ValidationError(_("You can't create past training!"))
+            if rec.training_start_date > rec.training_end_date:
+                raise ValidationError(
+                    _("End Date should be greater than Start date of Training!")
+                )
+            if rec.date_of_arrival and rec.date_of_arrival < rec.training_start_date:
```
Comment (MalayKhamar-SerpentCS) [inline hr_applicant/models/training.py]: Please check the condition of arrival date. It is not matching with the validation message.

File: hr_applicant/models/hr_emp_rel_edu.py [hunk 14]
```diff
@@ -0,0 +1,131 @@
...
+            self.birthday = False
+            return {"warning": warning}
+
+    @api.onchange("relative_type")
+    def _onchange_relative_type(self):
+        if self.relative_type:
+            self.gender = ""
+            if self.relative_type in ("Brother", "Father", "Husband", "Son", "Uncle"):
+                self.gender = "Male"
+            elif self.relative_type in ("Mother", "Sister", "Wife", "Aunty"):
```
Comment (MalayKhamar-SerpentCS) [inline hr_applicant/models/hr_emp_rel_edu.py]: What about "Daughter" ? Add it for Female Gender.

File: hr_applicant/report/report_view.xml [hunk 15]
```diff
@@ -0,0 +1,28 @@
...
+        <field name="margin_top">5</field>
+        <field name="margin_bottom">20</field>
+        <field name="margin_left">7</field>
+        <field name="margin_right">7</field>
+        <field name="header_line" eval="False" />
+        <field name="header_spacing">5</field>
+        <field name="dpi">90</field>
+    </record>
+    <record id="action_applicant_profile_form_report" model="ir.actions.report">
+        <field name="name">Applicant</field>
```
Comment (MalayKhamar-SerpentCS) [inline hr_applicant/report/report_view.xml:19]: Suggestion : Please print and check the pdf report format once.

**Review by MalayKhamar-SerpentCS — CHANGES_REQUESTED** [review]

**Comment by MalayKhamar-SerpentCS** [comment]
@RohanChristianSCS Please check and update.

**Review by MalayKhamar-SerpentCS — APPROVED** [review]

**Comment by MalayKhamar-SerpentCS** [comment]
Ok Thanks @RohanChristianSCS . 
@JayVora-SerpentCS Ready for Proceed to the next step.

**Pull request merged**